test/net-test.js

```diff
@@ -0,0 +1,785 @@
+      check(pkt);
+
+      pkt = packets.BlockTxnPacket.decode(pkt.encode());
+      check(pkt);
+    });
+
+    it('getproof', () => {
+      const check = (pkt) => {
+        assert.equal(pkt.type, packets.types.GETPROOF);
+        assert.bufferEqual(pkt.root, Buffer.alloc(32, 0x01));
+        assert.bufferEqual(pkt.key, Buffer.alloc(32, 0x02));
+      };
+
+      let pkt = new packets.GetProofPacket(Buffer.alloc(32, 0x01), Buffer.alloc(32, 0x02));
+
+      check(pkt);
+
+      pkt = packets.GetProofPacket.decode(pkt.encode());
+
+      check(pkt);
+    });
+
+    it('proof', () => {
+      const root = Buffer.alloc(32, 0x00);
+      const key = Buffer.alloc(32, 0x01);
+      const proof = new UrkelProof();
+
+      const check = (pkt) => {
+        assert.equal(pkt.type, packets.types.PROOF);
+        assert.bufferEqual(pkt.root, root);
+        assert.bufferEqual(pkt.key, key);
+        assert.bufferEqual(pkt.proof.encode(blake2b, 256), proof.encode(blake2b, 256));
+      };
+
+      let pkt = new packets.ProofPacket(root, key, proof);
+      check(pkt);
+
+      pkt = packets.ProofPacket.decode(pkt.encode());
+      check(pkt);
+    });
+
+    it('claim', () => {
+      const claim = new Claim();
+      const check = (pkt) => {
+        assert.equal(pkt.type, packets.types.CLAIM);
+        assert.bufferEqual(pkt.claim.encode(), claim.encode());
+      };
+
+      let pkt = new packets.ClaimPacket();
+      check(pkt);
+
+      pkt = packets.ClaimPacket.decode(pkt.encode());
+      check(pkt);
+    });
+
+    it('airdrop', () => {
+      const rawProof = read(AIRDROP_PROOF_FILE);
+      const proof = AirdropProof.decode(rawProof);
+
+      const check = (pkt) => {
+        assert.equal(pkt.type, packets.types.AIRDROP);
+        assert.bufferEqual(pkt.proof.encode(), proof.encode());
+      };
+
+      let pkt = new packets.AirdropPacket(proof);
+      check(pkt);
+
+      pkt = packets.AirdropPacket.decode(pkt.encode());
+      check(pkt);
+    });
+
+    it('unknown', () => {
+      const check = (pkt) => {
+        assert.equal(pkt.type, packets.types.UNKNOWN);
+        assert.bufferEqual(pkt.data, Buffer.alloc(12, 0x01));
+      };
+
+      let pkt = new packets.UnknownPacket(packets.types.UNKNOWN, Buffer.alloc(12, 0x01));
+      check(pkt);
+
+      pkt = packets.UnknownPacket.decode(pkt.encode(), packets.types.UNKNOWN);
+      check(pkt);
+    });
+  });
+
+  describe('Framer', function() {
+    it('will construct with network (primary)', () => {
+      const framer = new Framer();
+      assert.strictEqual(framer.network, Network.get('main'));
+    });
+
+    it('will construct with network (custom)', () => {
+      const framer = new Framer('regtest');
+      assert.strictEqual(framer.network, Network.get('regtest'));
+    });
+
+    it('throw with long command', () => {
+      const framer = new Framer('regtest');
+      let err = null;
+
+      try {
+        framer.packet('abcdefghijklm', Buffer.alloc(2, 0x00));
```

Comment:
Good catch, will update

Reply:
> We should also assert the upper bound on the size of the payload.

To do this, it would require allocating a buffer of ~4 GB.

It is only possible to allocate a buffer of ~2 GB using `Buffer.alloc` because of the max buffer length constant.

https://nodejs.org/api/buffer.html#buffer_buffer_constants_max_length

This value is not possible to alter, an error is thrown when it is altered.

I feel like 4 GB max size leaves open a dos vector where large packets are spammed and held in memory concurrently, causing an oom error.